src/lightning/pytorch/utilities/data/fileio.py

```diff
@@ -0,0 +1,83 @@
+import os
+import random
+import time
+from typing import Dict, Optional
+
+
+def is_url(path: str) -> bool:
+    return path.startswith("s3://")
+
+
+def is_path(path: str) -> bool:
+    return not is_url(path) and path.startswith("/")
+
+
+def path_to_url(path: str, bucket_name: str, bucket_root_path: str = "/") -> str:
+    if not path.startswith(bucket_root_path):
+        raise ValueError(f"Cannot create a path from {path} relative to {bucket_root_path}")
+    return f"s3://{bucket_name}/{os.path.relpath(path, bucket_root_path)}"
```

Comment:
we should double check if all of these are actually used.

Reply:
good point, removing